### DISCUSSION TITLE
Next Js Blog Most Read Posts from Google Analytics

### DISCUSSION THREAD

**Discussion by babyPrince** [vercel/next.js#14720] [Help]
Hey there,

I'm pretty new to Next and trying to figure how to approach and create something similar to [https://github.com/BlackBoxVision/gatsby-source-analytics-most-read-posts](https://github.com/BlackBoxVision/gatsby-source-analytics-most-read-posts). 

I have Next hooked up to Sanity CMS for my blog and would like to get the 10 most read articles/posts on build based on GA data. 

Has anyone worked on something like this? 
Currently I don't have any code to share, just gathering information on what's possible.

Thanks in advance

**Comment by leerob** [comment]
I have the perfect article for you! https://leerob.io/blog/google-analytics-api-nextjs

This article takes you from start to finish for creating an API route to fetch information from Google Analytics API. 

```js
import { google } from 'googleapis';

let googleAuth;

export default async (_, res) => {
  googleAuth = new google.auth.GoogleAuth({
    credentials: {
      client_email: process.env.GOOGLE_CLIENT_EMAIL,
      client_id: process.env.GOOGLE_CLIENT_ID,
      private_key: process.env.GOOGLE_PRIVATE_KEY
    },
    scopes: ['https://www.googleapis.com/auth/analytics.readonly']
  });

  const analytics = google.analytics({
    auth,
    version: 'v3'
  });

  const startDate = req.query.startDate;
  const response = await analytics.data.ga.get({
    'end-date': 'today',
    ids: 'ga:ID',
    metrics: 'ga:pageviews',
    'start-date': startDate
  });

  return res.status(200).json({
    pageViews: response.data.rows[0][0]
  });
};
```

**Reply by babyPrince** [reply]
Thanks @leerob ! 
It's crazy how hard it is to find any article related to Analytics that is not "how to add GA to your next app" as result.. :)
